### DISCUSSION TITLE
Distribution of Investment Cost

### DISCUSSION THREAD

**Discussion by zsalimian** [iiasa/message_ix#429] [MESSAGE modeling]
Hi,
I need to distribute investment cost of a power plant project over its lifetime period. 
As I saw in the documents, MESSAGE-ix just consider the construction time factor to scale up the investment costs.
Would you please see the following example to check whether I understood the investment process in MESSAGE-ix correctly?
For instance, we have a 1MW combined cycled plant which will definitely  be constructed between 2025 and 2030. It will operate from 2031. Considering 600 $/kw as investment cost of it, its investment costs is around 600$. From the documentation I understood that the model consider 600 occur in 2030, which is the year before its operation. But as it was constructed during 5 years it will scale up the investment cost using interest rate and construction time factor. So the model consider 600*(1+0.1)^5 as investment cost in 2030 (interest rate=0.1). Is it correct?
I mean the  investment cost is considered just for 2030, which is the year before operation.In other words, it doesn't distribute the investment cost during the period. Is it right?
Thank you

**Comment by zsalimian1359** [comment]
Can anybody help me in interpreting investment cost distribution in MESSAGE?

**Reply by volker-krey** [reply]
Yes, your interpretation is correct. You can find the code that calculates the construction_time_factor [here](https://github.com/iiasa/message_ix/blob/7ee162891b2e9ae6fdbed58e63727955689961e6/message_ix/model/MESSAGE/scaling_investment_costs.gms#L62). In case you are interested in adding alternative options for distributing investment costs over the construction period, please consider submitting a pull request. It would thought be important for the code to be configurable. In other words, the default behavior should not change.

**Comment by zsalimian** [comment]
Thank you for your reply.
Before adding alternative options I would like to know why the model scale up the investment costs by considering construction time period?
I know that the model minimizes the present value of investment cost. For instance, suppose we have a 1MW combined cycled plant which will definitely be constructed between 2025 and 2030. It will operate from 2031 and the investment cost is equal to 600 in 2025. The present value of future investment cost in 2030 is 600$, and its future value would be 600*(1+0.1)^5. I know that the model minimizes the present value of investment costs. So, I could not understand why the model considers the future value of investment cost by using construction_time_factor in the objective function?I mean the investment cost in the objective function should be 600$, why do you consider its future value in the objective function?

Thanks in advance for your kind reply
Regards

**Comment by zsalimian** [comment]
In other words, why the model consider CAP_NEW*construction_time_factor in the objective function?It represents the future value of the investment cost not its present value

**Comment by zsalimian** [comment]
Dear @volker-krey,
Would you please see the above question?
Regards
